File: tools/make_component.py
#!/usr/bin/python3

import sys

COMPONENT_TEXT = """
#ifndef {1}_HPP
#define {1}_HPP

#include "Component.hpp"

class {0} : public Component {{
	public:
		{0}();
		~{0}();
		void Initialize() override;
		void Update([[maybe_unused]] float deltaTime) override;
		void Render() override;
}};

{0}::{0}(){{}}
~{0}::{0}(){{}}

#endif
"""

def print_usage():
    print("USAGE:\nmake_component [component name]");

def main():

    if len(sys.argv) < 2:
        print_usage()
        return
    
    comp_name = sys.argv[1]
    with open(comp_name + ".hpp", "w") as header_file:
        header_file.write(COMPONENT_TEXT.format(comp_name, comp_name.upper()))

if __name__ == '__main__':
    main()

File: tools/make_component.lua
#!/usr/bin/lua

if #arg < 1 then
    print("USAGE:\nmake_component [component name]")
end

local COMPONENT_STRING =
[[
    #ifndef {1}_HPP
    #define {1}_HPP

#include "Component.hpp"

class {0} : public Component {
	public:
		{0}();
		~{0}();
		void Initialize() override;
		void Update(float deltaTime) override;
		void Render() override;
};

{0}::{0}(){}
~{0}::{0}(){}

#endif
]]

local first_step = COMPONENT_STRING:gsub("{0}", arg[1])
local second_step = first_step:gsub("{1}", string.upper(arg[1]))

local header_file = io.open(arg[1] .. ".hpp", "w")
header_file:write(second_step)